docs/routing-logic.md

```diff
@@ -0,0 +1,114 @@
+**Trino Gateway documentation**
+
+<table>
+  <tr>
+    <td>
+      <img src="./assets/logos/trino-gateway-v.png"/>
+    </td>
+    <td>
+      <ul>
+        <li><a href="quickstart.md">Quickstart</a></li>
+        <li><a href="installation.md">Installation</a></li>
+        <li><a href="security.md">Security</a></li>
+        <li><a href="operation.md">Operation</a></li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li><a href="gateway-api.md">Gateway API</a></li>
+        <li><a href="resource-groups-api.md">Resource groups API</a></li>
+        <li><a href="routing-rules.md">Routing rules</a></li>
+        <li><a href="routing-logic.md">Routing logic</a></li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li><a href="design.md">Design</a></li>
+        <li><a href="development.md">Development</a></li>
+        <li><a href="release-notes.md">Release notes</a></li>
+        <li><a href="references.md">References</a></li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+# Routing Logic
+
+## Overview
+Incoming requests are evaluated by the Gateway to determine if they
+are a follow-up to a request previously handled by the Gateway. If the 
+Gateway determines that they are, they are forwarded to whatever the 
+backend that handled the previous request.
+
+If the request does not have an antecedent, the Gateway first uses [Routing Rules](routing-rules.md)
+to determine which logical group of clusters, termed a "Routing Group",
+should handle the request. A member of the Routing Group, termed a backend, 
+is selected to handle the request using either an adaptive or round-robin 
+strategy.
+
+![Request Routing Flow](assets/gateway-routing-flow.svg)
+
+## Sticky Routing
+A request related to an ongoing process or state maintained on a single 
+backend cluster must be routed to that backend for proper handling. Two 
+mechanisms for identifying related requests are currently implemented.
+
+### Query ID Based Routing
+When a query is initiated through the Gateway, the query id will be 
+extracted from the response and mapped to the backend that provided the 
+response. Any subsequent request containing that query id will be forwarded 
+to that backend. For example, to retrieve query results, the trino client 
+polls a URI of the form 
+`v1/statement/executing/queryid/nonce/counter`. The Gateway will extract
+the queryid from this URI.
+
+### Cookie Based Routing
+The Gateway can add a session cookie to identify related requests. This is 
+useful for exchanges such as the OAuth handshake, where it isn't possible 
+to determine which backend handled the previous request based solely on
+the request URI and body.
+
+#### Cookie Based Routing Configuration
+`cookiePaths`: If the request URI starts with a path in this list, add a 
+session cookie if none exists. If a cookie exists, route the request to
+the backend associated with that cookie.
+
+`removeCookiePaths`: If the request URI starts with a path in this list,
```

Comment:
why do you need that? In what situation should a user configure that .. to what value

Reply:
Why would you need user to configure this? Other than logout?